ui/src/services/ApiClient.js

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios';
+
+class ApiClient {
```

Comment:
:+1: I like that one :)

Is there a way you could reduce the size for definitions of `get`/`post`/...? Maybe something like:

```javascript
  async send(request) {
    const {
        method = 'get',
        endpoint,
        payload = {},
        headers = {},
        params
    } = request;

    try {
        const response = await axios({
        method,
        headers: { ...this.headers, ...headers },
        params,
        url: endpoint,
        data: payload,
        ...this.settings
        });
        return response.data;
    } catch (error) {
        return { error };
    }
  }

  get = (...request) => send({ ...request, method: 'get' })

  // ...
```

Reply:
Let's try it

Reply:
Issue created #1338